app/src/androidTest/java/org/simple/clinic/bloodsugar/BloodSugarRepositoryAndroidTest.kt

```diff
@@ -48,22 +48,23 @@ class BloodSugarRepositoryAndroidTest {
 
   @Test
   fun saving_a_blood_sugar_reading_should_work_correctly() {
-    //given
-    val bloodSugarReading = BloodSugarReading(value = 10, type = Random)
+    // given
+    val now = Instant.now(clock)
+    val bloodSugarReading = BloodSugarReading(value = 50, type = Random)
     val patientUuid = UUID.fromString("a5921ec9-5c70-421a-bb0b-1291364683f6")
     val user = testData.qaUser()
 
-    //when
-    val testObserver = repository.saveMeasurement(
+    // when
+    val savedBloodSugar = repository.saveMeasurement(
         reading = bloodSugarReading,
         patientUuid = patientUuid,
         loggedInUser = user,
         facility = testData.qaFacility(),
-        recordedAt = Instant.now(clock)
-    ).test()
+        recordedAt = now)
+        .blockingGet()
 
-    //then
-    testObserver.assertComplete().assertNoErrors()
+    // then
+    assertThat(savedBloodSugar.recordedAt).isEqualTo(now)
```

Comment:
We should assert the complete `BloodSugarMeasurement` value and not just the `recorded` at timestamp.

Reply:
We can change the `BloodSugarRepository#saveMeasurement` function to accept the `UUID` as a parameter and then provide a `UUID` here to actually verify the complete measurement.